lib/xlat_tables/xlat_tables_common.c

```diff
@@ -284,8 +284,9 @@ static mmap_region_t *init_xlation_table_inner(mmap_region_t *mm,
 	unsigned level_size_shift = L1_XLAT_ADDRESS_SHIFT - (level - 1) *
 						XLAT_TABLE_ENTRIES_SHIFT;
 	unsigned level_size = 1 << level_size_shift;
-	unsigned long long level_index_mask = XLAT_TABLE_ENTRIES_MASK <<
-							level_size_shift;
+	unsigned long long level_index_mask =
+		((unsigned long long) XLAT_TABLE_ENTRIES_MASK)
+		<< level_size_shift;
```

Comment:
I don't understand what's happening, the integration and master branches have been synced
but this pull request is still showing 3 commits... I've created a new pull request, see #607,
as I couldn't figure out how to fix it. The code is exactly the same as this pull request. Sorry
for the confusion.